spec/acceptance/vpc_spec.rb

```diff
@@ -0,0 +1,106 @@
+require 'spec_helper_acceptance'
+require 'securerandom'
+
+describe "ec2_vpc" do
+
+  before(:all) do
+    @default_region = 'sa-east-1'
+    @aws = AwsHelper.new(@default_region)
+  end
+
+  def finder(name, method)
+    items = @aws.send(method, name)
+    expect(items.count).to eq(1)
+    items.first
+  end
+
+  def find_vpc(name)
+    finder(name, 'get_vpcs')
+  end
+
+  def find_dhcp_option(name)
+    finder(name, 'get_dhcp_options')
+  end
+
+  def find_route_table(name)
+    finder(name, 'get_route_tables')
+  end
+
+  def find_subnet(name)
+    finder(name, 'get_subnets')
+  end
+
+  def find_vpn_gateway(name)
+    finder(name, 'get_vpn_gateways')
+  end
+
+  def find_internet_gateway(name)
+    finder(name, 'get_internet_gateways')
+  end
+
+  def find_customer_gateway(name)
+    finder(name, 'get_customer_gateways')
+  end
+
+  def find_vpn(name)
+    finder(name, 'get_vpn')
+  end
+
+  describe 'should create a new VPC environment' do
+
+    before(:all) do
+      @name = "#{PuppetManifest.env_id}-#{SecureRandom.uuid}"
+      @config = {
+        :name => @name,
+        :ensure => 'present',
+      }
+
+      template = 'vpc.pp.tmpl'
+      @exit = PuppetManifest.new(template, @config).apply[:exit_status]
+    end
+
+    after(:all) do
+      new_config = @config.update({:ensure => 'absent'})
+      template = 'vpc_delete.pp.tmpl'
+      PuppetManifest.new(template, new_config).apply
+    end
+
+    it 'should run successfully with changes' do
+      expect(@exit.exitstatus).to eq(2)
+    end
```

Comment:
I've added a whole bunch more acceptance tests and assertions, including this check, in a future commit.